### PR TITLE
 fix(html): skip whitespace-only, template-like, and absolute URLs in source attributes

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

Four URL-skip divergences from html-loader found during the audit:
- Whitespace-only `src="   "` emitted a build error instead of being silently skipped
- Template expressions like `src="${foo}.png"` caused "Module not found" instead of being left untouched
- Absolute `http://` and `https://` URLs were rewritten to `data:,` via the scheme handler instead of passed through unchanged
- Protocol-relative `//example.com/` URLs had the same rewrite bug as http/https

All four are fixed with 3 lines in the source-walk loop, matching the style of the existing `#` check on the line above.

**What kind of change does this PR introduce?**

fix

**Did you add tests for your changes?**

Yes — `test/configCases/html/skip-behavior/`.

**Does this PR introduce a breaking change?**

No. These were error/incorrect-rewrite cases that now correctly pass through.

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**

n/a

**Use of AI**

PARITAL

### CODE REVIEW

File: lib/html/HtmlParser.js [hunk 1]
```diff
@@ -1212,6 +1212,8 @@ class HtmlParser extends Parser {
 				for (const parsedSource of parsedAttributeValue) {
 					const [value, innerStart, innerEnd] = parsedSource;
 					if (value.startsWith("#")) continue;
+					if (value.includes("${")) continue;
+					if (/^https?:\/\//i.test(value) || value.startsWith("//")) continue;
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/html/HtmlParser.js]: Remove this, we support http and https with experimental options, so we should support it

File: lib/html/HtmlParser.js [hunk 2]
```diff
@@ -1212,6 +1212,8 @@ class HtmlParser extends Parser {
 				for (const parsedSource of parsedAttributeValue) {
 					const [value, innerStart, innerEnd] = parsedSource;
 					if (value.startsWith("#")) continue;
+					if (value.includes("${")) continue;
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/html/HtmlParser.js]: Avoid it, it is legacy thing, we don't need this more, in future we implement simple template language to inject dot env variables

File: lib/html/HtmlParser.js [hunk 3]
```diff
@@ -1152,7 +1152,7 @@ class HtmlParser extends Parser {
 				if (!sourceItem) continue;
 
 				const attributeValue = attr.value;
-				if (!attributeValue) continue;
+				if (!attributeValue || !attributeValue.trim()) continue;
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/html/HtmlParser.js]: Let's fix it

**Comment by aryanraj45** [comment]
@alexander-akait done removed  the http/https and ${} skips, and  Switched to non-allocating !/\S/.test(). please take a look thanks !

**Review by alexander-akait — APPROVED** [review]

**Comment by aryanraj45** [comment]
@alexander-akait just checked The whitespace fix also affects the srcset test-it currently expects whitespace-only values to throw an error. Should I update the test and drop those 8 expectations or shall we close this PR in this favour ?

**Comment by alexander-akait** [comment]
Let's fix both variants here

**Comment by aryanraj45** [comment]
@alexander-akait now``` yarn test:base ./test/ConfigTestCases.basictest.js -t srcset --no-verbose``` passes vocally as well thanks
